test/kaniko_task_test.go

```diff
@@ -60,24 +61,50 @@ func getGitResource(namespace string) *v1alpha1.PipelineResource {
 	}
 }
 
-func getTask(namespace string, t *testing.T) *v1alpha1.Task {
+func getDockerRepo() (string, error) {
 	// according to knative/test-infra readme (https://github.com/knative/test-infra/blob/13055d769cc5e1756e605fcb3bcc1c25376699f1/scripts/README.md)
-	// the KO_DOCKER_REPO will be set with according to the porject where the cluster is created
-	// it is used here to dunamically get the docker registery to push the image to
+	// the KO_DOCKER_REPO will be set with according to the project where the cluster is created
+	// it is used here to dynamically get the docker registry to push the image to
 	dockerRepo := os.Getenv("KO_DOCKER_REPO")
 	if dockerRepo == "" {
-		t.Fatalf("KO_DOCKER_REPO env variable is required")
+		return "", fmt.Errorf("KO_DOCKER_REPO env variable is required")
 	}
+	return fmt.Sprintf("%s/kanikotasktest", dockerRepo), nil
+}
+
+func createSecret(c *knativetest.KubeClient, namespace string) (bool, error) {
+	// when running e2e in cluster, this will not be set so just hop out early
+	file := os.Getenv("KANIKO_SECRET_CONFIG_FILE")
```

Comment:
assuming we go ahead with this, can you add some docs to `test/README.md` on how to set this and what it should be?